### PR TITLE
Implement biz lic app renew service

### PR DESCRIPTION
**Pull request by ruaraya**
Implement business licence application renew service. The modification includes:

- Implementation of business licence application renewal process
- Add condition to validate "applicant is business manager" flag as a mandatory field
- Adjust validation of business insurance document
- Refactoring

### CODE REVIEW

File: src/Spd.Manager.Licence/BizLicAppMananger.cs [hunk 1]
```diff
@@ -202,4 +256,76 @@ private async Task<Unit> UpdateMembersAsync(Members members, Guid bizId, Guid ap
...
+            !existingFileInfos.Any(f => f.LicenceDocumentTypeCode == LicenceDocumentTypeCode.BizBranding))
+        {
+            throw new ApiException(HttpStatusCode.BadRequest, "Missing branding file.");
+        }
+
+        if (request.NoBranding == false &&
+            (newFileInfos.Count(f => f.LicenceDocumentTypeCode == LicenceDocumentTypeCode.BizBranding) +
+            existingFileInfos.Count(f => f.LicenceDocumentTypeCode == LicenceDocumentTypeCode.BizBranding) > 10))
+        {
+            throw new ApiException(HttpStatusCode.BadRequest, "Maximum of 10 documents allowed for branding was exceded.");
```
Comment (carolcarpenter) [inline src/Spd.Manager.Licence/BizLicAppMananger.cs:300]: should be 'exceeded'

Comment (ruaraya) [reply src/Spd.Manager.Licence/BizLicAppMananger.cs]: Nice catch, thanks!

File: src/Spd.Manager.Licence/BizLicAppMananger.cs [hunk 2]
```diff
@@ -112,7 +113,60 @@ public async Task<BizLicAppCommandResponse> Handle(BizLicAppReplaceCommand cmd,
...
+        // Validation: check if original licence meet renew condition.
+        LicenceListResp originalLicences = await _licenceRepository.QueryAsync(
+            new LicenceQry() { LicenceId = request.OriginalLicenceId },
+            cancellationToken);
+        if (originalLicences == null || !originalLicences.Items.Any())
+            throw new ArgumentException("cannot find the licence that needs to be renewed.");
+        LicenceResp originalLic = originalLicences.Items.First();
+
+        // Check Renew your existing application before it expires, within 90 days of the expiry date.
+        if (DateTime.UtcNow < originalLic.ExpiryDate.AddDays(-Constants.LicenceWith123YearsRenewValidBeforeExpirationInDays).ToDateTime(new TimeOnly(0, 0))
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/BizLicAppMananger.cs:129]: Have you check the specific requirment for this?

Comment (ruaraya) [reply src/Spd.Manager.Licence/BizLicAppMananger.cs:129]: I did, in confluence: https://justice.gov.bc.ca/wiki/display/SPDBT/Business+Licence

**Section: Licence Term**

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**